api/src/test/java/com/netflix/iceberg/transforms/TestTransformSerialization.java

```diff
@@ -80,4 +81,19 @@ public void testTransforms() throws Exception {
     Assert.assertEquals("Deserialization should produce equal partition spec",
         spec, TestHelpers.roundTripSerialize(spec));
   }
+
+  @SuppressWarnings("unchecked")
+  @Test(expected = IllegalArgumentException.class)
+  public void testApplyUnknownTransform() {
```

Comment:
Use `TestHelpers.assertThrows` instead of `expected`. It allows you to validate the exception message.